sys/include/net/eui48.h

```diff
@@ -56,6 +56,27 @@ static inline void eui48_to_eui64(eui64_t *eui64, const eui48_t *addr)
     eui64->uint8[7] = addr->uint8[5];
 }
 
+/**
+ * @brief   Generates an EUI-48 from a 64-bit device address
+ *
+ * @warning The resulting EUI-48 is not guaranteed to be unique.
+ *
+ * @param[out] eui48    the resulting EUI-48.
+ * @param[in]  addr     a 64-bit device address
+ */
+static inline void eui64_to_eui48(eui48_t *eui48, const eui64_t *addr)
+{
+    /* Preserve vendor id */
```

Comment:
Sorry for stepping in late, but shouldn't it be tested somewhere that the middle bytes of the EUI-64 are `ff:fe`? Otherwise, the EUI-64 isn't convertible to an EUI-48 AFAIK.

Reply:
See https://www.avrfreaks.net/comment/2614676#comment-2614676

Reply:
Isn't that conversion deprecated anyway? IMO, we should just mark the generated EUI48 as only locally unique. That would be no harm even if the middle bytes aren't `ff:fe`, or am I missing something?

Reply:
Well it looks like people will do this regardless, so we might as well give them a convenience function to do it in the most effective way :wink: 

https://github.com/RIOT-OS/RIOT/pull/15083#issuecomment-698579403

There is a warning that the resulting EUI is not globally unique, still better than relying on luid.

Reply:
> Isn't that conversion deprecated anyway?

Huh the up-conversion? Do you have a source for that?

Reply:
> There is a warning that the resulting EUI is not globally unique, still better than relying on luid.

Mh... I'm not convinced. There is still a chance that this way, the generated EUI-64 for the IPv6 address from the EUI-48 is the same as the original EUI-64, resulting in two identical addresses for the device. Rather using a completely different address seems more reliable (maybe using the original EUI-64 as additional "entropy")

Reply:
Wouldn't just marking the EUI-48 as only locally unique already result in different IPv6 addresses, assuming the EUI-64 was globally unique?

Reply:
> > Isn't that conversion deprecated anyway?
> 
> Huh the up-conversion? Do you have a source for that?

See https://standards.ieee.org/content/dam/ieee-standards/standards/web/documents/tutorials/eui.pdf:

> #### Mapping an EUI-48 to an EUI-64
> Mapping an EUI-48 to an EUI-64 is deprecated. The mapping is described here for historical reasons.

To my understanding, mapping is bidirectional - so the deprecation is also for both directions. So the conversion is to my understanding on the "dark side of the force" regardless of the `ff:fe` being present.

Reply:
> Wouldn't just marking the EUI-48 as only locally unique already result in different IPv6 addresses, assuming the EUI-64 was globally unique?

That's true... Mh...

Reply:
Ok, then NVM (still want a source for the deprecation of the up-conversion though :P)

Reply:
> Ok, then NVM (still want a source for the deprecation of the up-conversion though :P)

See https://github.com/RIOT-OS/RIOT/pull/15096#discussion_r496584515 ;-)

Reply:
Oops, somehow missed that. Sorry.

Reply:
> Oops, somehow missed that. Sorry.

Happens to me all the time. I also often review out-of date code in github until I notice a tiny red warning at the top indicating the code displayed is out of date. I think we should blame GitHub for not refreshing properly :-D